### PR TITLE
Update gradle depencies and versions.

### PR DESCRIPTION
**Pull request by IgorGanapolsky**
Update android sdk and lib versions.

### CODE REVIEW

File: examples/adapterExample/build.gradle [hunk 1]
```diff
@@ -2,13 +2,13 @@ apply plugin: 'com.android.application'
...
 android {
-    compileSdkVersion 19
-    buildToolsVersion "20.0.0"
+    compileSdkVersion 23
+    buildToolsVersion "23.0.0"
 
     defaultConfig {
         applicationId "io.realm.examples.realmadapters"
         minSdkVersion 15
-        targetSdkVersion 19
```
Comment (bmunkholm) [inline examples/adapterExample/build.gradle]: 123 -> 23

**Comment by cmelchior** [comment]
Hi @IgorGanapolsky
Thank you for updating our example dependencies. Keeping them up to date with the latest SDK is always good.

However updating our internal dependencies is out of the scope of this as we only update them if needed. But if you revert those, then this looks pretty good. Note there is also an example in `/experiments` which this PR also seems to have missed (and it broke our CI :) ).

File: build.gradle [hunk 2]
```diff
@@ -3,12 +3,12 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:1.2.3'
-        classpath 'de.undercouch:gradle-download-task:1.0'
-        classpath 'com.neenbedankt.gradle.plugins:android-apt:1.4'
-        classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.2'
+        classpath 'com.android.tools.build:gradle:1.3.0'
+        classpath 'de.undercouch:gradle-download-task:2.0.0'
```
Comment (cmelchior) [inline build.gradle]: Revert this please

File: build.gradle [hunk 3]
```diff
@@ -3,12 +3,12 @@ buildscript {
...
+        classpath 'com.neenbedankt.gradle.plugins:android-apt:1.7'
```
Comment (cmelchior) [inline build.gradle]: Revert this please

File: build.gradle [hunk 4]
```diff
@@ -3,12 +3,12 @@ buildscript {
...
+        classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.3.1'
         classpath 'com.jakewharton.sdkmanager:gradle-plugin:0.12.0'
-        classpath 'com.novoda:gradle-android-command-plugin:1.3.0'
+        classpath 'com.novoda:gradle-android-command-plugin:1.5.0'
```
Comment (cmelchior) [inline build.gradle]: Revert this please

**Comment by IgorGanapolsky** [comment]
So you are against updating to android library v23?

**Comment by cmelchior** [comment]
No, updating to v23 is :+1: 
It was the updates to the Realm build dependencies I was worried about

**Comment by cmelchior** [comment]
Closing due to no activity + it has been superseded by the new build system which upgraded these.

**Pull request closed without merging**